.github/workflows/ui.yml

```diff
@@ -0,0 +1,32 @@
+name: UI checks
+
+on: [push]
+
+jobs:
+ ui_job:
+    runs-on: ubuntu-latest
+    name: UI integrity check
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ github.ref }}
+      - name: Clean yarn.lock private dependencies
+        run: |
+          echo "Checking for private yarn.lock dependencies"
+          if grep -E "nexus\.vanti\.co\.uk/repository/npm-public/" ui/yarn.lock; then
+            echo "Private URLs found in yarn.lock"
+            echo "Auto cleaning yarn.lock dependencies"
+            echo "YARN_LOCK_CLEANED=true" >> $GITHUB_ENV
+            sed -i -e 's/nexus\.vanti\.co\.uk\/repository\/npm-public/registry\.npmjs\.org/g' ui/yarn.lock
+          else
+            echo "No private URLs found."
+          fi
+      - name: Commit & push auto-edited yarn.lock file
+        if: ${{ env.YARN_LOCK_CLEANED }}
+        run: |
+          git config user.name github-actions
+          git config user.email github-actions@github.com
+          git add ui/yarn.lock
+          git commit -m "[auto] fix: remove private repository references from ui/yarn.lock"
```

Comment:
I don't think we need both `[auto]` and `fix:` - I'd be happy with just `[auto]`